libdevcore/CMakeLists.txt

```diff
@@ -12,7 +12,7 @@ set(sources
 )
 
 add_library(devcore ${sources})
-target_link_libraries(devcore PRIVATE jsoncpp ${Boost_FILESYSTEM_LIBRARIES} ${Boost_REGEX_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(devcore PUBLIC jsoncpp ${Boost_FILESYSTEM_LIBRARIES} ${Boost_REGEX_LIBRARIES} ${Boost_SYSTEM_LIBRARIES} Threads::Threads)
```

Comment:
Should have been ``PUBLIC`` before already - the library is static and transitively depends on jsoncpp and the boost libs - anything linking against libdevcore will also have to link against these.